host/frontend/host-orchestrator/errors.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"net/http"
+)
+
+type AppError struct {
+	Msg        string
+	StatusCode int
+	Err        error
+}
+
+func (e *AppError) Error() string {
+	if e.Err != nil {
+		return e.Msg + ": " + e.Err.Error()
+	}
+	return e.Msg
+}
+func (e *AppError) Unwrap() error {
+	return e.Err
+}
+
+func (e *AppError) JSONResponse() ErrorMsg {
+	// Include only the high level error message in the error response, the
+	// lower level errors are just for logging
+	return ErrorMsg{Error: e.Msg}
+}
+
+func NewNotFoundError(msg string, e error) error {
```

Comment:
This one and the one in line 51 are never used, they should only be added when needed.

Reply:
removed, done